### PR TITLE
Document auto release procedure

### PR DESCRIPTION
**Pull request by mkolesnik**
This adds initial documentation of how to do semi-automated release.

Signed-off-by: Mike Kolesnik <mkolesni@redhat.com>

### CODE REVIEW

File: src/content/for_developers/release-process/automated.en.md [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+
+Sometimes you may want to create a specific project release for testing prior to creating a release candidate. In this
+case, the suggested naming convention is to append `-preN`.
+
+## Create Submariner product release
+
+The following sections outline the steps to be taken in order to create a full Submariner product release. As an example,
+we'll use version `v0.5.0`.
+
+The release process is mostly automated and is centered on a release YAML file which is constantly updated.
```
Comment (mangelajo) [inline src/content/for_developers/release-process/automated.en.md:44]:
```suggestion
The release process is mostly automated and is centered on a set of release YAMsL file which are constantly added.
```
? something like that? :)

File: src/content/for_developers/release-process/automated.en.md [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+
+   ```bash
+   ln -s <your-release-file.yaml> target
+   ```
+
+5) Commit the new files and create a new pull request.
+
+### Step 2: Create release for the `admiral` project
+
+Once the Shipyard pull request to pin the Admiral project is merged, use its commit id to pin to Admiral.
```
Comment (mangelajo) [inline src/content/for_developers/release-process/automated.en.md:68]: I don't understand this paragraph ':D

Comment (tpantelis) [reply src/content/for_developers/release-process/automated.en.md:68]: yeah - it seems we need the same steps to create the PR to pin as we do in the original doc.

File: src/content/for_developers/release-process/automated.en.md [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+3) Fill in the general fields for the release.
+
+4) Create a link to the file to make it the target release (So CI knows what we're releasing):
+
+   ```bash
+   ln -s <your-release-file.yaml> target
+   ```
+
+5) Commit the new files and create a new pull request.
+
```
Comment (tpantelis) [inline src/content/for_developers/release-process/automated.en.md:65]: What happens now? I assume CI runs to create the shipyard release? Do we merge the PR?

File: src/content/for_developers/release-process/automated.en.md [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+### Step 2: Create release for the `admiral` project
+
+Once the Shipyard pull request to pin the Admiral project is merged, use its commit id to pin to Admiral.
+
+1) Create a new branch, or use your branch from before, on the fork of releases repository.
```
Comment (tpantelis) [inline src/content/for_developers/release-process/automated.en.md:70]: I think we should recommend one or the other.

File: src/content/for_developers/release-process/automated.en.md [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+
+2) Update the `status` field to `admiral`.
+
+3) Add the `admiral` component with the commit hash to the release yaml.
```
Comment (tpantelis) [inline src/content/for_developers/release-process/automated.en.md:74]: Show the YAML code as well.

File: src/content/for_developers/release-process/automated.en.md [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+
+4) Commit the modified file and create a new pull request.
```
Comment (tpantelis) [inline src/content/for_developers/release-process/automated.en.md:76]: As before, what happens now? I assume CI runs to create the shipyard release? Do we merge the PR?

File: src/content/for_developers/release-process/automated.en.md [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+
+Once the pull requests to pin Lighthouse and Submariner to Admiral are merged, use their commit ids.
+
+1) Create a new branch, or use your branch from before, on the fork of releases repository.
+
+2) Update the `status` field to `projects`.
+
+3) Add the `submariner` and `lighthouse` components with the commit hashes to the release yaml.
+
+4) Commit the modified file and create a new pull request.
```
Comment (tpantelis) [inline src/content/for_developers/release-process/automated.en.md:88]: Similar comments as prior steps...

File: src/content/for_developers/release-process/automated.en.md (hunk 1 above)
Comment (tpantelis) [reply src/content/for_developers/release-process/automated.en.md:44]: I think: _The release process is mostly automated and uses a YAML file created in the  [releases](https://github.com/submariner-io/releases) repository that describes the release. This file is updated for each step in the release process._

File: src/content/for_developers/release-process/automated.en.md [hunk 8]
```diff
@@ -0,0 +1,189 @@
...
+case, the suggested naming convention is to append `-preN`.
+
+## Create Submariner product release
+
+The following sections outline the steps to be taken in order to create a full Submariner product release. As an example,
+we'll use version `v0.5.0`.
+
+The release process is mostly automated and is centered on a release YAML file which is constantly updated.
+On each step, once your pull request is reviewed and merged, the CI will create the required pull requests for
+other projects and they in turn would need to be reviewed and merged.
```
Comment (tpantelis) [inline src/content/for_developers/release-process/automated.en.md:46]:
```suggestion
Once the changes for a step are reviewed and merged, a CI job will run and create the required pull requests for the next step to be reviewed and merged.
```

**Review by tpantelis — CHANGES_REQUESTED** [review]

**Comment by mkolesnik** [comment]
Thanks for the reviews and suggestion, I've updated the document and incorporated many of them and hopefully it's much clearer now.

**Review by tpantelis — APPROVED** [review]
I pushed a commit with my changes/additions based on my experience going thru the process (definitely much easier than before 👍 ). I also pushed a commit to replace the manual version with the automated version. I don't think there's any reason to keep the manual version, especially since it differs from the automated process in that the latter creates the product release on the releases project.

File: src/content/development/release-process/_index.en.md [hunk 9]
```diff
@@ -31,179 +31,177 @@ The Dapper base image that is provided by `shipyard` for building and E2E tests
...
-5) Enter information to describe the release. This is optional for pre-releases but should be filled in for
-   a final release.
+Navigate to the [releases](https://github.com/submariner-io/releases) repository and fork it.
 
-6) If this is not a final release, mark the checkbox `This is a pre-release`. This includes release candidates. This is
-   important so it is not labeled as the Latest release in GitHub.
+1) Create a new branch for the intended release.
 
-7) Click `Publish release`.
+2) Create a new file in the `releases` directory (you can copy the `example.yaml file`). For our example. we'll name it `v0.8.0.yaml`.
```
Comment (nyechiel) [inline src/content/development/release-process/_index.en.md:65]:
```suggestion
2) Create a new file in the `releases` directory (you can copy the `example.yaml file`). For our example, we'll name it `v0.8.0.yaml`.
```

**Review by nyechiel — APPROVED** [review]

**Pull request merged**